src/v/storage/segment_reader.cc

```diff
@@ -50,6 +51,7 @@ segment_reader::~segment_reader() noexcept {
 
 segment_reader::segment_reader(segment_reader&& rhs) noexcept
   : _filename(std::move(rhs._filename))
+  , _open_lock(std::move(rhs._open_lock))
```

Comment:
this should probably be a different commit? is this also fixing a bug?

Reply:
Good call, this could be pulled up to a preceding commit.

Reply:
Good C++ question too. Still a noob so correct me if I'm wrong: Previously there was no move of the semaphore member `_open_lock`, which could cause the moved-from state to be lost, replaced in the destination object with default initialization?

Reply:
i think this change needs to be explained. moving the semaphore is not safe if there are any outstanding semaphore::units allocated, for example, as presumably the moved-from object is going to disappear.  A starting point would be to examine under which circumstances the segment reader is moved. it could be that the only real-world affect here is moving the name, but in that case, we could reinitialize the semaphore with the proper name.

Reply:
Oh I see what you mean. Agreed, will address in next update.